airflow/jobs/scheduler_job.py

```diff
@@ -1131,6 +1134,44 @@ def _emit_pool_metrics(self, session: Session = None) -> None:
     @provide_session
     def heartbeat_callback(self, session: Session = None) -> None:
         Stats.incr('scheduler_heartbeat', 1, 1)
+        if self.terminating:
+            return
+        if not self.is_alive():
+            self.on_kill()
```

Comment:
```suggestion
            self.on_kill()
```

Should this call `.kill()` instead?